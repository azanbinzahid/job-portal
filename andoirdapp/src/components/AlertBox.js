import React from 'react';
import {connect} from 'react-redux';
import {Text, Badge} from 'native-base';

const AlertBox = (props) => {
  if (props.alerts !== null && props.alerts.length > 0) {
    return (
      <>
        {props.alerts.map((alert) => (
          <Badge key={alert.id} info rounded>
            <Text>{alert.msg}</Text>
          </Badge>
        ))}
      </>
    );
  } else {
    return null;
  }
};

const mapStateToProps = (state) => ({
  alerts: state.alertReducer.alerts,
});

export default connect(mapStateToProps)(AlertBox);
