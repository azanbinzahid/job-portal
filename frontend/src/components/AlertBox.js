import React from 'react'
import {useSelector} from 'react-redux';
import { Alert } from 'react-bootstrap';


const AlertBox = () => {

    const alertReducer = useSelector(state => state.alertReducer)
    const alerts = alertReducer.alerts
   
    if(alerts !==null && alerts.length>0){
        let alertsList;
        alertsList = alerts.map((alert)=>{
            return(
                <Alert key={alert.id} variant={alert.alertType}>
                    {alert.msg}
                </Alert>
            )
        });

        return alertsList;
    } else {
        return null
    }
}

export default AlertBox;