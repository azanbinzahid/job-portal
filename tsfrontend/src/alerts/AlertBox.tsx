import React, { FC } from "react";
import { connect } from "react-redux";
import { Toast } from "react-bootstrap";
import { Alert as AlertType } from "redux/types";
import { RootState } from "redux/reducers";

type Props = {
  alerts: AlertType[];
};

function getColor(alertType: String) {
  let alertColor = "blue";
  if (alertType === "danger") {
    alertColor = "red";
  } else if (alertType === "warning") {
    alertColor = "grey";
  } else {
    alertColor = "green";
  }
  return alertColor;
}

export const AlertBox: FC<Props> = (props) => {
  if (props.alerts !== null && props.alerts.length > 0) {
    return (
      <>
        {props.alerts.map((alert) => (
          <Toast key={alert.id}>
            <Toast.Header>
              <strong className="mr-auto">Notification </strong>
            </Toast.Header>
            <Toast.Body
              style={{
                fontWeight: "bold",
                color: getColor(alert.alertType),
              }}
            >
              {alert.msg}
            </Toast.Body>
          </Toast>
        ))}
      </>
    );
  } else {
    return null;
  }
};

const mapStateToProps = (state: RootState) => ({
  alerts: state.alertReducer.alerts,
});

export default connect(mapStateToProps)(AlertBox);
