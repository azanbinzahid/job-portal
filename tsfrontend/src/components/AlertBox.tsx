import React, { FC } from "react";
import { connect } from "react-redux";
import { Alert } from "react-bootstrap";
import { Alert as AlertType } from "redux/types";
import { RootState } from "redux/reducers";

type Props = {
  alerts: AlertType[];
};

const AlertBox: FC<Props> = (props) => {
  if (props.alerts !== null && props.alerts.length > 0) {
    return (
      <>
        {props.alerts.map((alert) => (
          <Alert key={alert.id} variant={alert.alertType}>
            {alert.msg}
          </Alert>
        ))}
      </>
    );
  } else {
    return null;
  }
};

const mapStateToProps = (state: RootState) => ({
  alerts: state.alertReducer.alerts,
});

export default connect(mapStateToProps)(AlertBox);
