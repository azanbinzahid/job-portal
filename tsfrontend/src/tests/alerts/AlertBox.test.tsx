import React from "react";
import { shallow } from "enzyme";
import { AlertBox } from "alerts/AlertBox";
test("renders the component", () => {
  const component = shallow(<AlertBox alerts={[]} />);
  expect(component).toMatchSnapshot();
});
