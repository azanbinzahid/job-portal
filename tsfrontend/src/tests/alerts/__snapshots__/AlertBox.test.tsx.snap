// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Fragment>
  <Toast
    key="abc"
  >
    <ToastHeader
      closeButton={true}
      closeLabel="Close"
    >
      <strong
        className="mr-auto"
      >
        Notification 
      </strong>
    </ToastHeader>
    <ToastBody
      style={
        Object {
          "color": "red",
          "fontWeight": "bold",
        }
      }
    >
      danger
    </ToastBody>
  </Toast>
</Fragment>
`;

exports[`renders the component 2`] = `""`;
